util/rejectWhenEmptyArray.js

```diff
@@ -0,0 +1,15 @@
+import InternalError from '@/errors/InternalError';
+
+/**
+ * Rejects the promise if the data is an empty array.
+ *
+ * @param {Promise} promise
+ * @returns {Promise}
+ */
+export async function rejectWhenEmptyArray(promise) {
+  const data = await promise;
+  if (!data || data.length === 0) {
+    throw new InternalError('Empty data');
```

Comment:
To um pouco na dúvida se isso deveria ser um internal error nesse ponto do código mesmo ou lidar com isso lá na ponta na hora de lidar com a resposta 👀

Reply:
Coloquei aqui para evitar duplicação de código, mas se tiver alguma outra ideia estou aberto a sugestões